### PR TITLE
Only extern "C" when compiled with c++ compiler

### PR DESCRIPTION
**Pull request by yujincheng08**

### CODE REVIEW

File: arghandling.o
Comment (thunder-coding) [inline arghandling.o:1]: You are not supposed to put binary blobs in git history. Probably you should be using a separate build directory outside source tree

File: arghandling.c [hunk 1]
```diff
@@ -16,7 +16,9 @@
    by the number of arguments used.
    Too bad we can't just use getopt for all of this, but we don't have
    enough information to do it right.  */
-
+#ifdef __cplusplus
```
Comment (thunder-coding) [inline arghandling.c:19]: Isn't this source file written in C, there is no need to use `exterm "C"` here (unless you are trying to C source files with a C++ compiler)

File: arghandling.o
Comment (yujincheng08) [reply arghandling.o:1]: oh sorry i didnt notice i missly uploaded it

File: arghandling.h [hunk 2]
```diff
@@ -1,5 +1,8 @@
 /* Taken from emacs */
 #define ARRAYELTS(arr) (sizeof (arr) / sizeof (arr)[0])
 
-extern "C" bool
+#ifdef __cplusplus
```
Comment (thunder-coding) [inline arghandling.h:4]: This looks good, although most compilers don't need it, and will simply ignore the `extern "C"` thing.

**Review by thunder-coding — CHANGES_REQUESTED** [review]
Hey thanks for the PR, I've requested a few changes. Please go through them

File: arghandling.c (hunk 1 above)
Comment (yujincheng08) [reply arghandling.c:19]: yes, sometimes you may try to compile it with c++ compiler

**Comment by yujincheng08** [comment]
removed missly uploaded blob file

File: arghandling.h (hunk 2 above)
Comment (yujincheng08) [reply arghandling.h:4]: but for c++ compiler it will mangle the name and throw a link error

File: arghandling.c (hunk 1 above)
Comment (thunder-coding) [reply arghandling.c:19]: > yes, sometimes you may try to compile it with c++ compiler.

Understandible, but I don't think this workaround is good for code quality. Are there any other projects which use similar hacks?

> for example, you want to compile it in one line of command.

Maybe just consider taking a few seconds and use autotools to setup the build for you. We also have a few tests, autotools should make it easier to execute them.

Comment (yujincheng08) [reply arghandling.c:19]: aha, magisk uses it. See https://github.com/topjohnwu/Magisk/pull/6759/files#diff-b70bfb027a8ef3ec4ab07a53a3b8ba83e63334ba3ce5ca6d9d773900ed452e5eR198

Comment (yujincheng08) [reply arghandling.c:19]: autotool is good but I dont know if macOS allows us to do so.

**Pull request closed without merging**